source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,284 @@
+===========
+Search Text
+===========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you will learn how to run a **text search** in MongoDB.
+
+In MongoDB, you can use a text search to retrieve documents that
+contain a term or a phrase in a specified field. A term is a sequence of
+characters that excludes whitespace characters. A phrase is a sequence
+of terms that with any number of whitespace characters.
+
+Sample Documents
+~~~~~~~~~~~~~~~~
+
+In the following sections, you run text searches on the
+``fast_and_furious_movies`` collection. Each section uses ``collection``
+to refer to the ``MongoCollection`` instance of the
+``fast_and_furious_movies`` collection. 
+
+This collection contains the following documents that represent one of
+the several movies that are part of the Fast and Furious movie franchise
+and tags they are mostly associated with:
+
+.. code-block:: json
+
+   { "_id": 1, "title": "2 Fast 2 Furious ", "tags": ["undercover", "drug dealer"] } 
+   { "_id": 2, "title": "Fast 5", "tags": ["bank robbery", "full team"] }
+   { "_id": 3, "title": "Furious 7", "tags": ["emotional"] }
+   { "_id": 4, "title": "The Fate of the Furious", "tags": ["betrayal"] }
+
+Text Index
+----------
+
+You must create a **text index** before running a text search. A text
+index specifies the string or string array field for the query to run a
+text search on. A collection can only contain one text index.
+
+In the following examples, you run text searches on the ``title``
+field in the ``fast_and_furious_movies`` collection. To enable text
+searches on the ``title`` field, create a text index using the
+:ref:`Indexes <index-text-indexes>` builder with the following 
+snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+.. note::
+
+   If you want to run text searches on both the ``title`` and
+   ``tags`` field in the ``fast_and_furious_movies`` collection, you
+   need to create an text index to include them.
+
+   If you already have an existing text index, you need to drop the
+   index by specifying the existing text index field name with the
+   following snippet: 
+
+   .. code-block:: java
+
+      collection.dropIndex("field_name");
```

Comment:
If you specify the process of dropping a text index by its field name, it's not clear to me how to drop a text index with multiple field names. Could you describe that somewhere? Or otherwise how to determine the name of the index?

Reply:
I mention in the sentence after you specify a *

Reply:
That drops _all_ indexes including non-text indexes. I don't think this is a practical solution because you generally have other meaningful indexes on a collection to improve search/sort speed.

Reply:
I'm removing this part and redirecting it to the Indexes page.